File: src/models/provider.rs
use serde::{Deserialize, Serialize};
use chrono::{DateTime, Utc};
use crate::models::ModelInfo;

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ProviderInfo {
    pub provider: String,
    pub provider_name: String,
    pub last_updated: DateTime<Utc>,
    pub models: Vec<ModelInfo>,
}

impl ProviderInfo {
    pub fn new(provider: String, provider_name: String, models: Vec<ModelInfo>) -> Self {
        Self {
            provider,
            provider_name,
            last_updated: Utc::now(),
            models,
        }
    }
}

File: src/models/provider.ts
import { ModelInfo } from './model_info';

export interface ProviderInfo {
  provider: string;
  providerName: string;
  lastUpdated: string; // ISO 8601 string format for DateTime<Utc>
  models: ModelInfo[];
  api?: string;
  doc?: string;
  description?: string;
  metadata?: Record<string, unknown>;
  tags?: string[];
}

export function createProviderInfo(
  provider: string,
  providerName: string,
  models: ModelInfo[],
  overrides: Partial<Omit<ProviderInfo, 'provider' | 'providerName' | 'models'>> = {},
): ProviderInfo {
  const { lastUpdated = new Date().toISOString(), ...rest } = overrides;

  return {
    provider,
    providerName,
    models,
    lastUpdated,
    ...rest,
  };
}
